cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/TetheringAgentService.java

```diff
@@ -183,14 +213,11 @@ protected long runTask() {
     }
 
     // Update last message ids in the store for all peers
-    TransactionRunners.run(transactionRunner, context -> {
-      AppMetadataStore appMetadataStore = AppMetadataStore.create(context);
-      for (Map.Entry<String, String> entry : lastMessageIds.entrySet()) {
-        appMetadataStore.persistSubscriberState(entry.getKey(), SUBSCRIBER, entry.getValue());
-      }
-    });
-
-    return connectionInterval;
+    AppMetadataStore appMetadataStore = getAppMetadataStore(context);
```

Comment:
don't need to do this anymore, the superclass calls storeMessageId() for you

Reply:
see previous comment about lastMessageIds